zio-query/shared/src/main/scala/zio/query/ZQuery.scala

```diff
@@ -558,6 +594,30 @@ final class ZQuery[-R, +E, +A] private (private val step: ZIO[(R, QueryContext),
         case (_, Result.Fail(e))                                => Result.fail(e)
       }
     }
+
+  /**
+   * Returns a query that models the execution of this query and the specified
+   * query in parallel, combining their results with the specified function.
+   * Requests composed with `zipWithParQuery` or combinators derived from it
+   * will automatically be batched.
+   *
+   * Unlike `zipWithPar`, `zipwithParQuery` will not perform arbitrary effects
+   * in parallel but will only batch requests to data sources. If you do not
+   * need to perform arbitrary effects in parallel this can be significantly
+   * more efficient.
+   */
+  final def zipWithParQuery[R1 <: R, E1 >: E, B, C](that: ZQuery[R1, E1, B])(f: (A, B) => C): ZQuery[R1, E1, C] =
+    ZQuery {
+      self.step.zipWith(that.step) {
+        case (Result.Blocked(br1, c1), Result.Blocked(br2, c2)) => Result.blocked(br1 && br2, c1.zipWithPar(c2)(f))
```

Comment:
I see you added `zipWithParQuery` in `Continue`, it should be called here 😄

Reply:
Done!